src/main/java/co/rsk/federate/FederatorSupport.java

```diff
@@ -261,6 +257,43 @@ public Optional<Address> getProposedFederationAddress() {
             .map(addr -> Address.fromBase58(getBtcParams(), addr));
     }
 
+    public Optional<Integer> getProposedFederationSize() {
+        BigInteger size = bridgeTransactionSender.callTx(
+            federatorAddress, Bridge.GET_PROPOSED_FEDERATION_SIZE);
+
+        return Optional.ofNullable(size)
+            .map(BigInteger::intValue);
+    }
+
+    public Optional<ECKey> getProposedFederatorPublicKeyOfType(int index, FederationMember.KeyType keyType) {
+        Objects.requireNonNull(keyType);
+
+        byte[] publicKeyBytes = bridgeTransactionSender.callTx(
+            federatorAddress,
+            Bridge.GET_PROPOSED_FEDERATOR_PUBLIC_KEY_OF_TYPE,
+            new Object[]{ index, keyType.getValue() });
+       
+        return Optional.ofNullable(publicKeyBytes)
+            .map(ECKey::fromPublicOnly);
+    }
+
+    public Optional<Instant> getProposedFederationCreationTime() {
+        BigInteger creationTime = bridgeTransactionSender.callTx(
+            federatorAddress, Bridge.GET_PROPOSED_FEDERATION_CREATION_TIME);
+
+        return Optional.ofNullable(creationTime)
+            .map(BigInteger::longValue)
+            .map(Instant::ofEpochMilli);
```

Comment:
Let's agree on this @julia-zack and @jeremy-then. Mili or seconds?

Reply:
I think we should go with seconds yes